src/main/java/core/basesyntax/stratege/PurchaseStrategy.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.stratege;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.TransactionDto;
+
+public class PurchaseStrategy implements OperationStrategy {
+    @Override
+    public void doOperation(TransactionDto transactionDto) {
+        if (transactionDto.getQuantity() < 0) {
+            throw new IllegalArgumentException("Purchase can't be negative number "
+                    + transactionDto.getQuantity());
+        }
+        if (!Storage.fruitsAndAmountsMap.containsKey(transactionDto.getFruit())) {
```

Comment:
it would be easier to read the code if we move `transactionDto.getFruit()` and `transactionDto.getQuantity()` to local variables